engine/access/rpc/backend/backend_stream_transactions.go

```diff
@@ -0,0 +1,205 @@
+	txExecuted         bool
+	lastTxStatus       flow.TransactionStatus
+}
+
+// SubscribeTransactionStatuses subscribes to transaction status changes starting from the transaction reference block ID.
+// If invalid tx parameters will be supplied SubscribeTransactionStatuses will return a failed subscription.
+func (b *backendSubscribeTransactions) SubscribeTransactionStatuses(ctx context.Context, tx *flow.TransactionBody) subscription.Subscription {
+	nextHeight, err := b.blockTracker.GetStartHeightFromBlockID(tx.ReferenceBlockID)
+	if err != nil {
+		return subscription.NewFailedSubscription(err, "could not get start height")
+	}
+
+	txInfo := TransactionSubscriptionMetadata{
+		txID:               tx.ID(),
+		txReferenceBlockID: tx.ReferenceBlockID,
+		messageIndex:       counters.NewMonotonousCounter(0),
+		blockWithTx:        nil,
+		blockID:            flow.ZeroID,
+		lastTxStatus:       flow.TransactionStatusUnknown,
+	}
+
+	sub := subscription.NewHeightBasedSubscription(
+		b.sendBufferSize,
+		nextHeight,
+		b.getTransactionStatusResponse(&txInfo),
+	)
+
+	go subscription.NewStreamer(b.log, b.broadcaster, b.sendTimeout, b.responseLimit, sub).Stream(ctx)
+
+	return sub
+}
+
+// getTransactionStatusResponse returns a callback function that produces transaction status
+// subscription responses based on new blocks.
+func (b *backendSubscribeTransactions) getTransactionStatusResponse(txInfo *TransactionSubscriptionMetadata) func(context.Context, uint64) (interface{}, error) {
+	return func(ctx context.Context, height uint64) (interface{}, error) {
+		highestHeight, err := b.blockTracker.GetHighestHeight(flow.BlockStatusFinalized)
+		if err != nil {
+			return nil, fmt.Errorf("could not get highest height for block %d: %w", height, err)
+		}
+
+		// Fail early if no block finalized notification has been received for the given height.
+		// Note: It's possible that the block is locally finalized before the notification is
+		// received. This ensures a consistent view is available to all streams.
+		if height > highestHeight {
+			return nil, fmt.Errorf("block %d is not available yet: %w", height, subscription.ErrBlockNotReady)
+		}
+
+		if txInfo.lastTxStatus == flow.TransactionStatusSealed || txInfo.lastTxStatus == flow.TransactionStatusExpired {
+			return nil, fmt.Errorf("transaction final status %s was already reported %w", txInfo.lastTxStatus.String(), subscription.ErrResponseNotAvailableForBlock)
+		}
+
+		if txInfo.blockWithTx == nil {
+			// Check if block contains transaction.
+			txInfo.blockWithTx, err = b.searchForTransactionBlock(height, txInfo)
+			if err != nil {
+				return nil, err
+			}
+		}
+
+		// Find the transaction status.
+		var txStatus flow.TransactionStatus
+		if txInfo.blockWithTx == nil {
+			txStatus, err = b.txLocalDataProvider.DeriveUnknownTransactionStatus(txInfo.txReferenceBlockID)
+		} else {
+			if txInfo.blockID == flow.ZeroID {
+				txInfo.blockID = txInfo.blockWithTx.ID()
```

Comment:
this should go immediately after setting `txInfo.blockWithTx`. no need to check it every block